Fix Go bindings possible nested comment warnings

bluk opened the pull request:
When using the Go bindings in a project, you can get compilation warnings for the `/*` in the comment blocks. These warnings seem unnecessary.

Warning is:

```
# github.com/apple/foundationdb/bindings/go/src/fdb
In file included from _cgo_export.c:4:
cgo-gcc-export-header-prolog:43:1: warning: '/*' within block comment [-Wcomment]
cgo-gcc-export-header-prolog:44:1: warning: '/*' within block comment [-Wcomment]
```

I wasn't too sure on the correct code style to fix these errors so if there's a different one, feel free to change or suggest differently.

These warnings can cause noise in IDEs that try to parse the error/warning messages during a build.

ajbeamon commented on bindings/go/src/fdb/fdb.go:
```diff
@@ -37,9 +37,9 @@ import (
-/* (https://code.google.com/p/go-wiki/wiki/cgo#Global_functions) */
+/* Would put this in futures.go but for the documented issue with */
+/* exports and functions in preamble                              */
+/* (https://code.google.com/p/go-wiki/wiki/cgo#Global_functions)  */
```
It looks to me like other functions are consistently commented with `//` starting every line. I'm not sure why we had commented this one with a different style, but probably the best change now is to make it match the other functions.

ajbeamon commented on bindings/go/src/fdb/database.go:
```diff
@@ -96,7 +96,7 @@ func retryable(wrapped func() (interface{}, error), onError func(Error) FutureNi
 		}
 
-		/* If OnError returns an error, then it's not
+		/* If OnError returns an error, then it's not       */
```
I'm inclined to say we should also use `//` for this comment, even though there are other cases (mostly just in this file) where we use `/* */` for comments on code. Possibly we could change the other single-line occurrences in this file as well if we wanted.

bluk commented:
I updated the changes to use the single line comments `//` throughout the file. Also, I rebased the changes onto the latest master. Sorry for the delay.

bluk commented on bindings/go/src/fdb/database.go:
```diff
@@ -22,10 +22,8 @@
- #include <foundationdb/fdb_c.h>
-*/
+// #define FDB_API_VERSION 610
+// #include <foundationdb/fdb_c.h>
```
No problem, I reverted the preambles for the changed files files (changing all of them seems to be wasteful noise IMO).
